ResearchKit/ActiveTasks/ORKWaitStepViewController.m

```diff
@@ -0,0 +1,218 @@
+    } else if (_activityIndicatorView) {
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:_activityIndicatorView attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:0.0]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-(>=0)-[_activityIndicatorView]-(>=0)-|" options:NSLayoutFormatAlignAllBaseline metrics:nil views:NSDictionaryOfVariableBindings(_activityIndicatorView)]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[_textLabel]-[_activityIndicatorView]|" options:NSLayoutFormatAlignAllCenterX metrics:nil views:NSDictionaryOfVariableBindings(_activityIndicatorView, _textLabel)]];
+    }
+}
+
+#pragma mark Accessibility
+
+- (BOOL)isAccessibilityElement {
+    return YES;
+}
+
+- (NSString *)accessibilityLabel {
+    if (_progressView) {
+        return ORKAccessibilityStringForVariables(_textLabel.accessibilityLabel, _progressView.accessibilityLabel);
+    } else if (_activityIndicatorView) {
+        return ORKAccessibilityStringForVariables(_textLabel.accessibilityLabel, _activityIndicatorView.accessibilityLabel);
+    }
+    return nil;
+}
+
+- (UIAccessibilityTraits)accessibilityTraits {
+    if (_progressView) {
+        return [super accessibilityTraits] | UIAccessibilityTraitUpdatesFrequently;
+    } else {
+        return [super accessibilityTraits];
+    }
+}
+
+@end
+
+@interface ORKWaitStepViewController ()
+
+@property (nonatomic, strong) ORKWaitView *waitView;
+
+@end
+
+@implementation ORKWaitStepViewController {
+    ORKProgressIndicatorMask _indicatorMask;
+}
+
+- (instancetype)initWithStep:(ORKStep *)step {
+    self = [super initWithStep:step];
+    if (self) {
+        self.suspendIfInactive = NO;
+    }
+    return self;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.learnMoreButtonItem = nil;
+    self.activeStepView.headerView.captionLabel.text = nil;
+
+    _waitView = [[ORKWaitView alloc] initWithIndicatorMask:((ORKWaitStep *)self.step).indicatorMask heading:self.step.title];
+    _waitView.translatesAutoresizingMaskIntoConstraints = NO;
+    self.activeStepView.activeCustomView = _waitView;
+    if (((ORKWaitStep *)self.step).shouldContinueOnFinish) {
+        self.activeStepView.continueSkipContainer.hidden = YES;
+    } else {
+        self.activeStepView.continueSkipContainer.continueEnabled = false;
+    }
+    
+}
+
+- (void)stepDidChange {
+    [super stepDidChange];
+    if (_waitView) {
+        [_waitView removeFromSuperview];
+    }
+    
+    _waitView = [[ORKWaitView alloc] initWithIndicatorMask:((ORKWaitStep *)self.step).indicatorMask heading:self.step.title];
+    _waitView.translatesAutoresizingMaskIntoConstraints = NO;
```

Comment:
Don't remove it from superview if it exists. Only update the values for it. See example below:

```
    if (_waitView) {
        _waitView.indictatorMask = ((ORKWaitStep *)self.step).indicatorMask;
        _waitView.textLabel.text = self.step.title;
    } else {
        _waitView = [[ORKWaitView alloc] initWithIndicatorMask:((ORKWaitStep *)self.step).indicatorMask heading:self.step.title];
        _waitView.translatesAutoresizingMaskIntoConstraints = NO;
    }
```